test/gtest/check_names.py

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env python3
+###############################################################################
+#
+# MIT License
+#
+# Copyright (c) 2024 Advanced Micro Devices, Inc.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+#################################################################################
+
+"""gtest name linter"""
+import re
+import sys
+import argparse
+
+"""regexp based on https://github.com/ROCm/MIOpen/wiki/GTest-development#naming"""
+re_prefix = re.compile(r"^((Smoke)|(Full)|(Perf)|(Unit.*))")
+re_hw = re.compile(r"^((CPU)|(GPU))")
+re_datatype = re.compile(
+    r"^((FP((8)|(16)|(32)|(64)))|(BFP((8)|(16)))|(I((8)|(32)))|(NONE))"
+)
+
+
+def parse_args():
+    """Function to parse cmd line arguments"""
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        "--list",
+        dest="list",
+        type=str,
+        required=True,
+        help="Specify gtest test list file",
+    )
+    args = parser.parse_args()
+
+    return args
+
+
+def main():
+    """Main function"""
+    args = parse_args()
+    mismatched_prefix = []
+    mismatched_hw = []
+    mismatched_datatype = []
+
+    with open(args.list) as fp:
```

Comment:
It might be a good idea to wrap this into a function then call the function into main like you did with parse_args()